timesketch/lib/datastores/opensearch.py

```diff
@@ -1271,3 +1296,505 @@ def flush_queued_events(self, retry_count=0):
+                    os_logger.error(
+                        (
+                            "[Slice %s/%s] 'sort' field missing in last hit. "
+                            "Cannot continue with search_after for this slice."
+                        ),
+                        log_slice_id,
+                        num_slices,
+                    )
+                    stop_event.set()
+                    break
+
+        except NotFoundError:
+            os_logger.warning(
+                "[Slice %s/%s] PIT ID %s not found or expired, "
+                "or an index in the list was not found.",
+                log_slice_id,
+                num_slices,
+                pit_id if pit_id else "N/A",
+            )
+            stop_event.set()  # Signal issue
+        except Exception as e:  # pylint: disable=broad-exception-caught
+            os_logger.error(
+                "[Slice %s/%s] An unexpected error occurred: %s",
+                log_slice_id,
+                num_slices,
+                str(e),
+                exc_info=True,
+            )
+            stop_event.set()  # Signal failure
+        finally:
+            if pit_id:
+                try:
+                    self.client.delete_pit(body={"pit_id": [pit_id]})
+                    os_logger.info(
+                        "[Slice %s/%s] Deleted PIT ID: %s",
+                        log_slice_id,
+                        num_slices,
+                        pit_id,
+                    )
+                except Exception as e:  # pylint: disable=broad-exception-caught
+                    log_msg = "[Slice %s/%s] Error deleting PIT ID %s: %s"
+                    os_logger.error(
+                        log_msg,
+                        log_slice_id,
+                        num_slices,
+                        pit_id,
+                        str(e),
+                        exc_info=True,
+                    )
+
+            os_logger.info(
+                "[Slice %s/%s] Finished. Processed %s documents.",
+                log_slice_id,
+                num_slices,
+                total_docs_in_slice,
+            )
+            # Signal this worker is done by putting a sentinel, even if an error
+            # occurred (unless queue itself failed)
+            try:
+                output_queue.put(
+                    None, timeout=5
+                )  # Short timeout, main thread should be consuming
+            except queue.Full:
+                os_logger.error(
+                    "[Slice %s/%s] Failed to put sentinel: Queue full.",
+                    log_slice_id,
+                    num_slices,
+                )
+            except Exception as e_s:  # pylint: disable=broad-exception-caught
+                os_logger.error(
+                    "[Slice %s/%s] Failed to put sentinel: %s",
+                    log_slice_id,
+                    num_slices,
+                    str(e_s),
+                )
+
+    def export_events_with_slicing(
```

Comment:
a general question, if this is used on a busy system, how would one see which slice log relates to which? Would it make sense to create uuid() in any way and add it to every log messa written in one export job?

Reply:
Each slice log (if enabled in the opensearch.py - by default those logs are not visible) is listing its slice ID in the front. e.g. `"[Slice %s/%s] Retrieved %s docs. Total for this slice so far: %s."`.
And if used with the celery system for example they will be tied to the celery execution ID.

Reply:
ah okay, yeah then its fine